fladle-plugin/src/main/java/com/osacky/flank/gradle/FulladlePlugin.kt

```diff
@@ -24,115 +23,127 @@ class FulladlePlugin : Plugin<Project> {
+          if (project.isAndroidAppModule) {
+            // app modules produce app apks that we can consume
             flankGradleExtension.debugApk.set(rootProject.provider { this@app.outputFile.absolutePath })
-          } else {
-            // Otherwise, let's just add it to the list.
-            strs.add("- app: ${this@app.outputFile}")
-          }
-
-          // If the instrumentation apk isn't yet set, let's use this one.
-          if (!flankGradleExtension.instrumentationApk.isPresent) {
-            flankGradleExtension.instrumentationApk.set(rootProject.provider { this@test.outputFile.absolutePath })
-          } else {
-            // Otherwise, let's just add it to the list.
-            strs.add("      test: ${this@test.outputFile}")
+          } else if (project.isAndroidLibraryModule) {
+            // library modules do not produce an app apk and we'll use the one specified in fulladleModuleConfig block
+            // we need library modules to specify the app apk to test against, even if it's a dummy one
+            check(fulladleModuleExtension.debugApk.isPresent && fulladleModuleExtension.debugApk.orNull != null) {
+              "Library module ${project.path} did not specify a debug apk. Library modules do not generate a debug apk and one needs to be specified in the fulladleModuleConfig block\nThis is a required parameter in FTL which remains unused for library modules under test, and you can use a dummy apk here"
+            }
+            flankGradleExtension.debugApk.set(rootProject.provider { fulladleModuleExtension.debugApk.get() })
           }
-
-          if (strs.isEmpty()) {
-            // this is the root module
-            // should not be added as additional test apk
-            return@test
+        } else {
+          // Otherwise, let's just add it to the list.
+          if (project.isAndroidAppModule) {
+            strs.add(" app: ${this@app.outputFile}")
+          } else if (project.isAndroidLibraryModule) {
+            // app apk is not required for library modules so only use if it's explicitly specified
+            if (fulladleModuleExtension.debugApk.orNull != null) {
+              strs.add(" app: ${fulladleModuleExtension.debugApk.get()}")
+            }
           }
+        }
 
-          val maxTestShards = propertyToYaml(fulladleModuleExtension.maxTestShards, "max-test-shards")
+        // If the instrumentation apk isn't yet set, let's use this one.
+        if (!flankGradleExtension.instrumentationApk.isPresent) {
+          flankGradleExtension.instrumentationApk.set(rootProject.provider { this@test.outputFile.absolutePath })
+        } else {
+          // Otherwise, let's just add it to the list.
+          strs.add("      test: ${this@test.outputFile}")
+        }
 
-          val clientDetails = mapPropertyToYaml(fulladleModuleExtension.clientDetails, "client-details")
+        if (strs.isEmpty()) {
+          // this is the root module
```

Comment:
why do we need this? when executing fulladle, can we just always use the root values and when executing from a single module, can we always use the values set there?
otherwise, it should be an error to have both configured.